sdk/pinocchio/src/lib.rs

```diff
@@ -1,12 +1,222 @@
+//! * `instruction_data`: data for the instruction
+//!
+//! Pinocchio also offers variations of the program entrypoint
+//! ([`lazy_program_entrypoint`]) and global allocator ([`no_allocator`]). In
+//! order to use these, the program needs to specify the program entrypoint,
+//! global allocator and panic handler individually. The [`entrypoint!`] macro
+//! is equivalent to writing:
+//! ```ignore
+//! program_entrypoint!(process_instruction);
+//! default_allocator!();
+//! default_panic_handler!();
+//! ```
+//! Any of these macros can be replaced by other implementations and Pinocchio
+//! offers a couple of variants for this.
+//!
+//! ### [`lazy_program_entrypoint!`]
+//!
+//! The [`entrypoint!`] macro looks similar to the "standard" one found in
+//! [`solana-program`](https://docs.rs/solana-program-entrypoint/latest/solana_program_entrypoint/macro.entrypoint.html).
+//! It parsers the whole input and provides the `program_id`, `accounts` and
+//! `instruction_data` separately. This consumes compute units before the program
+//! begins its execution. In some cases, it is beneficial for a program to have
+//! more control when the input parsing is happening, even whether the parsing
+//! is needed or not &mdash; this is the purpose of the [`lazy_program_entrypoint!`]
+//! macro. This macro only wraps the program input and provides methods to parse
+//! the input on-demand.
+//!
+//! The [`lazy_program_entrypoint`] is suitable for programs that have a single
+//! or very few instructions, since it requires the program to handle the parsing,
+//! which can become complex as the number of instructions increases. For *larger*
+//! programs, the [`program_entrypoint!`] will likely be easier and more efficient
+//! to use.
+//!
+//! To use the [`lazy_program_entrypoint!`] macro, use the following in your
+//! entrypoint definition:
+//! ```ignore
+//! use pinocchio::{
+//!   default_allocator,
+//!   default_panic_handler,
+//!   entrypoint::InstructionContext,
+//!   lazy_program_entrypoint,
+//!   msg,
+//!   ProgramResult
+//! };
+//!
+//! lazy_program_entrypoint!(process_instruction);
+//! default_allocator!();
+//! default_panic_handler!();
+//!
+//! pub fn process_instruction(
+//!   mut context: InstructionContext
+//! ) -> ProgramResult {
+//!     msg!("Hello from my lazy program!");
+//!     Ok(())
+//! }
+//! ```
+//!
+//! The [`InstructionContext`](entrypoint::InstructionContext) provides on-demand
+//! access to the information of the input:
+//!
+//! * [`available()`](entrypoint::InstructionContext::available): number of available
+//!   accounts.
+//! * [`next_account()`](entrypoint::InstructionContext::next_account): parsers the
+//!   next available account (can be used as many times as accounts available).
+//! * [`instruction_data()`](entrypoint::InstructionContext::instruction_data): parsers
+//!   the intruction data.
+//! * [`program_id()`](entrypoint::InstructionContext::program_id): parsers the
```

Comment:
```suggestion
//! * [`program_id()`](entrypoint::InstructionContext::program_id): parses the
```

Reply:
I consistently made the same typo over multiple lines. 😅